mm2src/mm2_bitcoin/serialization/src/reader.rs

```diff
@@ -51,12 +51,14 @@ pub trait Deserializable {
 
 #[derive(Debug)]
 pub enum CoinVariant {
+    LBC,
```

Comment:
I don't like adding a new coin variant for every new supported coin if the block header deserialization needs it, I guess this is acceptable in this PR for now. But maybe in the future we can have something called `block_header_features` in the coin config that specifies additional requirements for block header deserialization, e.g. `metadata_trie_root: true` for LBC and so on. What do you think @borngraced @artemii235?

Reply:
I see `BlockHeader` and maybe other types as a part of some trait, e.g.:
```
pub trait UtxoCommonOps {
    type BlockHeader;
    type Transaction;
    ...
}
```
But in this case, we will likely have to add `LbcCoin` implementing this trait differently with a lot of boilerplate code (we can work around this with `Derive`, though).

> But maybe in the future we can have something called block_header_features in the coin config that specifies additional requirements for block header deserialization, e.g. metadata_trie_root: true for LBC and so on.

It can be `block_header_format` instead, mapping its value to a specific struct. Might be better than associated type approach I mentioned above.

@shamardy Could you please create an issue with this refactoring suggestion and ideas mentioned? 🙂

Reply:
> @shamardy Could you please create an issue with this refactoring suggestion and ideas mentioned? 🙂

Done